coba/experiments/tasks.py

```diff
@@ -160,10 +163,12 @@ def indexify(action,actions,discrete,batched):
                 reward   = rewards.eval(_action)
                 feedback = feedbacks.eval(_action) if feedbacks else reward
 
-                if record_time  : out['predict_time'] = predict_time
-                if record_prob  : out['probability']  = prob
-                if record_action: out['action']       = action
-                if feedbacks    : out['feedback']     = feedback
+                if record_time   : out['predict_time'] = predict_time
+                if record_prob   : out['probability']  = prob
+                if record_action : out['action']       = action
+                if record_actions: out['actions']      = actions
+                if feedbacks     : out['feedback']     = feedback
+                if record_rewards: out['rewards']      = rewards._values
```

Comment:
Is pulling out the `_values` not appropriate for the reward functions you mentioned?

Reply:
Yes and no. For continuous action problems the answer is definitely not. Continuous action problems aren't the norm though so I think we should be able to support this functionality in the case of discrete action problems and leave it for sometime in the future to support continuous actions better. So my thought is to do something like we did with 'rank'.

If you look up on line 112 there's this:

```
if 'rank' in self._record and not discrete:
    warnings.warn(f"The rank metric can only be calculated for discrete environments")
```

I think we should simply add another warning for `if 'rewards' in self._record and not discrete`.

Then the line that records rewards should look like this:

```
if record_rewards: out['rewards'] = list(map(rewards.eval,actions))
```

I'm happy to explain why but also you can just copy and paste too if you'd like :). Sometimes I get too pedantic.

Reply:
(Oh, and if you want to be an absolute hero, super extra credit, you could figure out how to get SimpleEvaluation to work with the `Logged` filter in `coba/environments/filters.py`. It's an interesting puzzle and I haven't been able to come up with a solution yet that doesn't require complete violation or destruction of the current module structure.)